packages/core/src/Stores/client-store.js

```diff
@@ -229,25 +229,13 @@ export default class ClientStore extends BaseStore {
 
     @computed
     get legal_allowed_currencies() {
-        if (!this.landing_companies) return [];
+        if (!this.landing_companies && !this.current_landing_company) return [];
```

Comment:
Second condition check `!this.current_landing_company` relies on the `landing_companies` value, if the first one is truthy then the second part will always be truthy. 

(nitpick) Let's move `(this.root_store.ui && this.root_store.ui.real_account_signup_target)` condition check into the above, so we can remove the nested if condition. 

```suggestion
        if (
            !this.landing_companies ||
            !(this.root_store.ui && this.root_store.ui.real_account_signup_target
        )    {
                return [];
        }
```